tests/system_test.py

```diff
@@ -488,6 +493,29 @@ def wait_ports(self, **retry_kwargs):
         wait_ports(self.ports_family, **retry_kwargs)
 
 
+class NginxServer(Process):
+    """
+    Sets up an ngix server
+    """
+    def __init__(self,
+                 config_path: str,  # Full path of templated config file (string) /blah-blah/nginx/nginx-template.conf
+                 env: dict[str, str],  # A dict of env variables which will be substituted in the config_path file.
+                 name: str="nginx",
+                 expect: int=Process.RUNNING,
+                 **kwargs):
+        input_file = os.path.join(config_path)
+        config_file = os.path.join(env['setupclass-folder'], 'nginx.conf')
+
+        # Run the expandvars.py script to replace the shell style variables in the config file with
+        # environment variables.
+        args = [sys.executable, os.path.join(expandvars_script_folder, script_file), input_file, config_file]
+        subprocess.run(args, stderr=STDOUT, check=True, env=env)
+
+        # Pass the config file for nginx to use.
+        nginx_cmd = ['nginx', '-c', config_file]
+        super(NginxServer, self).__init__(nginx_cmd, name=name, expect=expect, **kwargs)
```

Comment:
Just an observation, in python 3, you don't have to provide name of current class in super()

```
class C(B):
    def method(self, arg):
        super().method(arg)    # This does the same thing as:
                               # super(C, self).method(arg)
```

https://docs.python.org/3/library/functions.html#super

Reply:
Done !